File: frontend/src/types/payment.ts
export interface Payment {
  id: string;
  order_id: string;
  amount: number;
  status: string;
  payment_method?: string;

  // ePayco information
  epayco_transaction_id?: string;
  epayco_response_code?: number;
  epayco_response_message?: string;
  epayco_approval_code?: string;
  epayco_receipt?: string;

  // Payment method information
  card_last_four?: string;
  card_brand?: string;

  // Client information
  client_name?: string;
  client_email?: string;
  client_phone?: string;
  client_ip?: string;

  // Timestamps
  created_at: string;
  processed_at?: string;
  updated_at: string;

  // Order information
  order_total: number;
  order_status: string;
  order_created_at: string;
}

export interface PaymentsResponse {
  payments: Payment[];
  total: number;
}

export type PaymentStatus =
  | "completed"
  | "pending"
  | "failed"
  | "rejected"
  | "reversed"
  | "retained"
  | "started"
  | "expired"
  | "abandoned"
  | "canceled";

File: backend/app/schemas/payment.py
from datetime import datetime
from decimal import Decimal
from typing import Optional, List
from pydantic import BaseModel
from uuid import UUID

class PaymentBase(BaseModel):
    amount: Decimal
    status: str
    payment_method: Optional[str] = "credit_card"
    
class PaymentCreate(PaymentBase):
    order_id: UUID
    # ePayco information
    epayco_transaction_id: Optional[str] = None
    epayco_response_code: Optional[int] = None
    epayco_response_message: Optional[str] = None
    epayco_approval_code: Optional[str] = None
    epayco_receipt: Optional[str] = None
    
    # Payment method information
    card_last_four: Optional[str] = None
    card_brand: Optional[str] = None
    
    # Client information
    client_name: Optional[str] = None
    client_email: Optional[str] = None
    client_phone: Optional[str] = None
    client_ip: Optional[str] = None
    
    processed_at: Optional[datetime] = None

class PaymentOut(PaymentBase):
    id: UUID
    order_id: UUID
    
    # ePayco information
    epayco_transaction_id: Optional[str] = None
    epayco_response_code: Optional[int] = None
    epayco_response_message: Optional[str] = None
    epayco_approval_code: Optional[str] = None
    epayco_receipt: Optional[str] = None
    
    # Payment method information
    card_last_four: Optional[str] = None
    card_brand: Optional[str] = None
    
    # Client information
    client_name: Optional[str] = None
    client_email: Optional[str] = None
    client_phone: Optional[str] = None
    client_ip: Optional[str] = None
    
    # Timestamps
    created_at: datetime
    processed_at: Optional[datetime] = None
    updated_at: datetime

    class Config:
        from_attributes = True

class PaymentWithOrder(PaymentOut):
    """Payment with basic order information"""
    order_total: Decimal
    order_status: str
    order_created_at: datetime

class PaymentsResponse(BaseModel):
    payments: List[PaymentWithOrder]
    total: int 